File: 1000/2055B/main.java
import java.util.*;
 
public class main 
{
    public static void main(String[]args)
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while(t-- > 0)
        {
            int n = sc.nextInt(), a[] = new int[n], b[] = new int[n];

            for(int i=0;i<n;i++)
                a[i] = sc.nextInt();

            for(int i=0;i<n;i++)
                b[i] = a[i]-sc.nextInt();

            Arrays.sort(b);
            System.out.println(b[0]+b[1] >=0 ? "YES":"NO");
        }
    }
}

File: 1000/2055B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;

        vector<int> a(n), b(n);

        for (int i = 0; i < n; i++)
            cin >> a[i];

        for (int i = 0; i < n; i++) 
        {
            int x;
            cin >> x;
            b[i] = a[i] - x;
        }

        sort(b.begin(), b.end());

        if (b[0] + b[1] >= 0)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}
